### DISCUSSION TITLE
how to set sp for daily or minutes frequency data

### DISCUSSION THREAD

**Discussion by LishuaiJing3** [alan-turing-institute/sktime#1418] [Q&A]
Hi, I have time series that has daily pattern, every N minutes, for example N = 1 or N = 4, how to set the sp parameter for autoets or autoarima estimator?

**Comment by fkiraly** [comment]
`sp` should be the multiple of your observation frequency, I believe - e.g., if your data is per-minute, and you have a periodicity of 4 minutes, use `sp=4`.

**Reply by albertobenavides** [reply]
Are you sure? Check this: https://robjhyndman.com/hyndsight/seasonal-periods/

**Reply by fkiraly** [reply]
Ah, I misspoke when I said "multiple of observation frequency". What I said is wrong. I meant "multiple of you observation period length". Or "correspond to an integer fraction of your observation frequency".

Frequency is inverse of period length (times a constant depending on unit), so the above is not entirely accurate, mathematically speaking.

The formally correct answer should be:
`sp` should be a multiple of your (pattern/periodicity) period length, and the exact value should be (pattern/periodicity) period length in units of observation period length.

Example: if your data has index period of a minute (a `freq` of `min` or `T`), and the model periodicity should be 4min, then set `sp=4`.

Morally, the answer I gave above is still sort-of "right", if you gloss over the mis-use of the word "frequency" (which I think is in line with its mis-use in `pandas`, where actually it means period length, as integer multiples of `freq` objects double the period length and not the frequency).

**Reply by fkiraly** [reply]
in comparison to the Hyndman article: in `sktime`, `sp` is *multiple of observation period*, whereas it seems that `freq` for hyndman is *multiple of observation frequency*. So for Hyndman, `freq=24` with a reference of a day period means hourly, whereas in `sktime`, `sp=24` with a reference of a daily period would mean 24 day periods.

**Reply by fkiraly** [reply]
(we take the index period length as reference, so multiplying frequency would result in periods smaller than the minimum period step, which seems not very useful)

**Reply by albertobenavides** [reply]
We are using `sp` just as you described. I am glad to read your response. Thank you, @fkiraly :)

**Reply by fkiraly** [reply]
Great!

PS: if you would like to try out sth experimental, have a look at `PluginParamsForecaster` in `sktime.param_est.plugin` (since 0.13.1).

The docstring example shows how to plug in automatically inferred `sp` into any forecaster.
